lib/screens/contributors.dart

```diff
@@ -295,16 +356,50 @@ Future<List<ContributorCard>> getContributors(
   return contriCards;
 }
 
-addToContributors() {
-  // TODO: Trigger an alert box or something similar
-  // Ask the user to enter the details
-  // Adds the user to contributors list
+void addToContributors(BuildContext context,
+    Function(ContributorCard) addContributorCallback) async {
+  // Fetch contributor data from GitHub based on the provided username
+  String githubUsername =
+      'githubUsername'; // Replace with the actual GitHub username
+
+  // Fetch contributor data from GitHub using the GitHub API
+  final githubResponse =
+      await http.get(Uri.https('api.github.com', '/users/$githubUsername'));
+  if (githubResponse.statusCode == 200) {
+    // Parse the GitHub response to extract contributor details
+    final contributorData = contributorDetailFromJson(githubResponse.body);
+
+    ContributorCard newContributor = ContributorCard(
+      userName: contributorData.login,
+      name: contributorData.name ?? '',
+      desc: contributorData.bio ?? '',
+      location: contributorData.location ?? '',
+      twitterUsername: contributorData.twitterUsername ?? '',
+      displayImgUrl: contributorData.avatarUrl,
+      website: contributorData.blog.isEmpty
+          ? contributorData.htmlUrl
+          : contributorData.blog,
+    );
+
+    // Access cardList and update it within the _ContributorsState instance
+    addContributorCallback(newContributor);
+
+    try {

```

Comment:
Using BuildContexts across async functions is bad practice see if u can refactor code and avoid it